index.html

```diff
@@ -86,49 +138,211 @@
+      .append('g')
+      .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
 
     // SCALES
 
+    chart.x = d3.scale.log()
+          .base(1.5)
+          .domain(d3.extent(data, function (d) { return d.democratContribution; }))
+          .range([0, width])
+          .nice();
+
+    chart.y = d3.scale.log()
+          .base(1.5)
+          .domain(d3.extent(data, function (d) { return d.republicanContribution; }))
+          .range([height, 0])
+          .nice();
+
+
     // AXES
+    
+    formatAxis = d3.format(".2s");
 
-    // EVENT HANDLERS
+    var xAxis = d3.svg.axis()
+                  .scale(chart.x)
+                  .tickSize(-height)
+                  .orient("bottom")
+                  .tickFormat(formatAxis);
 
-    d3.select('#year-range').on('change', function () {
-      options.yearRange = d3.event.target.value;
-      update();
-    });
+    var yAxis = d3.svg.axis()
+                  .scale(chart.y)
+                  .tickSize(-width)
+                  .orient("left")
+                  .tickFormat(formatAxis);
+
+    chart.svg.append("g")
+       .attr("class", "axis") 
+       .attr("transform", "translate(0," + (height) + ")")
+       .call(xAxis)
+       .append("text")
+       .attr("class", "axis-title")
+       .attr("y", margin.bottom)
+       .attr("x", width/2)
+       .style("text-anchor", "middle")
+       .text("Total Contributions to Democrats");
+
+    chart.svg.append("g")
+       .attr("class", "axis") 
+       //.attr("transform", "translate(0,0)")
+       .call(yAxis)
+       .append("text")
+       .attr("class", "axis-title")
+       .attr("transform", "rotate(-90)")
+       .attr("y", -margin.left)
+       .attr("x", -(height/2))
+       .attr("dy", ".71em")
+       .style("text-anchor", "middle")
+       .text("Total Contributions to Republicans");;
+
+    chart.tooltip = d3.tip()
+                      .attr('class','tooltip')
+                      .html(function (d) {return 'Contributions to Democrats: $' + d3.format(",.2f")(d.democratContribution) })
+
+
+
+       chart.update();
+       chart.tooltip(chart.svg); 
   }
 
-  function update() {
+  Chart.prototype.update = function() {
+
+          var chart = this;
+
     // TRANSFORM
     txData = data.slice();
 
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+
+    if (options.state !== 'all') {
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
-      });
+        return d.state === options.state;
+      }); 
     }
 
     // SCATTERPLOT
+  
+
+    var points = chart.svg.selectAll('.point')
+                    .data(txData);
+
+                    points
+                    // This is what I would think would make them fade out but
+                    // it just breaks the code and nothing fades in:
+                    //.transition().duration(1000)
+                    .attr('r', 0);
```

Comment:
So this is the "update" selection — all the dots that currently exist and will continue to. By setting the radius to zero, the dots first disappear before reappearing, which probably isn't exactly the effect we're looking for.